py-polars/tests/unit/io/test_csv.py

```diff
@@ -1639,3 +1639,57 @@ def write(self, data: bytes) -> int:
     df.write_csv(buf)  # type: ignore[call-overload]
 
     assert b"".join(buf.writes) == b"a,b\n10,x\n20,y\n30,z\n"
+
+
+@pytest.mark.write_disk()
+def test_read_filelike_object(tmp_path: Path) -> None:
```

Comment:
Though better. I think we can do this test completely in memory by using `io.Buffer`

Reply:
Sorry for misunderstood. I confirmed that this issue is also caused with `io.BufferedReader` and this PR fixed it.
So I have committed the update.